### PR TITLE
Stuff

### PR DESCRIPTION
**Pull request by dirkschumacher**
Just playing a bit with the code.

* Only accept valid looking voting files in /votes
* Commit on submit. 

How could someone commit to the repository given the private key of the creator? I don't really see in the api how you can pass a private key or retrieve it.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -42,6 +41,10 @@ const patch = snabbdom.init([
 		votes.push(vote)
 	}
 
+	const commit = async () => {
+		await archive.commit()
```
Comment (derhuerst) [inline src/index.js]: Thanks, didn't know that we need to do this with the [`DatArchive` API](https://beakerbrowser.com/docs/apis/dat.html), as it mimics [the `hyperdrive` API](https://www.npmjs.com/package/hyperdrive) which doesn't have this.

File: src/index.js [hunk 2]
```diff
@@ -42,6 +41,10 @@ const patch = snabbdom.init([
 		votes.push(vote)
 	}
 
+	const commit = async () => {
```
Comment (derhuerst) [inline src/index.js]: Not sure a single `await` is really worth the async function. I'd rather use `async`/`await` below, in `onSubmit`. Not an issue though.

Comment (dirkschumacher) [reply src/index.js]: So you are saying to move it to `onSubmit`?

File: src/index.js (hunk 1 above)
Comment (dirkschumacher) [reply src/index.js]: Year previously, those changed only went into the staging and weren't commited

File: src/index.js (hunk 2 above)
Comment (derhuerst) [reply src/index.js]: yup. just `await archive.commit()` in `onSubmit`.

File: src/index.js [hunk 3]
```diff
@@ -68,21 +68,16 @@ const patch = snabbdom.init([
 		votes.push(vote)
 	}
 
-	const commit = async () => {
-		await archive.commit()
-	}
-
-	const onSubmit = (author, chosen) => {
-		addVote({
+	const onSubmit = async (author, chosen) => {
```
Comment (derhuerst) [inline src/index.js]: FYI if the promise returned by `onSubmit()` won't be `catch`ed.

**Pull request merged**